### ISSUE TITLE
Temporary file not deleted when reading mf4 with "unfinalized_standard_flags" == 1

### ISSUE THREAD

**Issue by timohencken** [danielhrisca/asammdf#531]
# Python version
('python=3.7.9 (tags/v3.7.9:13c94747c7, Aug 17 2020, 18:58:18) [MSC v.1900 64 bit (AMD64)]')
'os=Windows-10-10.0.18362-SP0'
'numpy=1.20.1'
'asammdf=6.2.0'

# Code 

  ## MDF version
  4.10
  * file must be >= version 4.10
  * file must have FileIndetificationBlock with "unfinalized_standard_flags" == 1
  * sample file attached: [zipped](https://github.com/danielhrisca/asammdf/files/6776827/stripped_sample.zip)
    * I could not attach the original file due to its contents and size, but I created a stripped one showing the same issue

  ## Code snippet
```
from asammdf import MDF
mdf_obj = MDF('stripped_sample.mf4')
del mdf_obj
# check files in /tmp
# -> stripped_sample.mf4 was created in tmp but not deleted
```
  
  ## Traceback
no traceback, as no Exception occurs
tracked down to following code: 
https://github.com/danielhrisca/asammdf/blob/31d0c2a9c9c013c3fec667d52339788e241444fa/asammdf/blocks/mdf_v4.py#L384-L390
  
# Description
If reading an MDF file with "unfinalized_standard_flags" == 1,
- the file is copied to /tmp before reading
- the file never gets deleted
==> (possibly big) file accumulate in /tmp

Posibble workarounds (that I can think of):

1. using tempfile.NamedTemporaryFile
  * PRO: will clean the file automatically on close()
  * CON: file cannot be accessed anymore after closed once
2. delete temporary file in __del__()
  * PRO: file will only be deleted after we are sure it will never be accessed anymore
  * CON: additional flag required to check if tmp file is in use (do NOT delete the original source file)
```
 def __init__():
+	self._tmp_file_used = False
 	[...]
 	if version >= "4.10" and flags:
 		[...]
+		self._tmp_file_used = True
 def __del__():
 	[...]
+	if self._tmp_file_used:
+		self.name.unlink()
```

BR
Timo

**Comment by danielhrisca** [comment]
Hello Timo,

thanks for the nice summary. I will look into this next week

**Comment by danielhrisca** [comment]
Hello Timo,

can you check the new development code?

**Comment by timohencken** [comment]
Hi Daniel,
thanks for the path.
I just checked it and 
* it works fine 👍 
* but has 2 drawbacks...

1. this would raise the minimum python version to 3.8
  * see `Changed in version 3.8: The missing_ok parameter was added` https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink
  * in order to keep min. python version at 3.6, we would need
```
            if Path(self.name).parent == Path(gettempdir()):
                try:
                    Path(self.name).unlink()  # missing_ok would require py 3.8
                except FileNotFoundError:
                    pass  # file already gone
```
2. if user is reading a file which is actually stored to /tmp, it would get deleted
* avoiding this would be a little complicated, as we cannot use the original filename for copying
* as the risk is quite low, I would leve it to you if you would like to work around or ignore this case
* one possibility would be to use a specific tag in the name to be sure we created the file - like this:
```
class MDF4:
    # ...
    def __init__(self, name=None, version="4.10", channels=(), **kwargs):
    # ...
                if version >= "4.10" and flags:
                    from tempfile import NamedTemporaryFile
                    tmp_file = NamedTemporaryFile(
                        suffix='.mf4',
                        prefix='__asammdf_tmp__',
                        delete=False
                    )
                    self.name = Path(tmp_file.name)
                    tmp_file.close()
                    shutil.copy(name, self.name)
                    self._file = open(self.name, "rb+")
                    self._from_filelike = False
                    self._read(mapped=False)

    def close(self):
    # ...
        if Path(self.name).parent == Path(gettempdir()) and Path(self.name).name.startswith('__asammdf_tmp__'):
            try:
                Path(self.name).unlink()  # missing_ok would require py 3.8
            except FileNotFoundError:
                pass  # file already gone
```

**Comment by danielhrisca** [comment]
I've made the change to keep Python 3.6 compatibility. I also don't want to use the ``__asammdf_tmp__`` prefix because it will not look nice in the GUI for that file tab.

**Comment by timohencken** [comment]
Hi Daniel, 
I finally found the time to create a PR, but I just saw that you already implemented my 1st proposal.
👍 Thanks for keeping the python version requirement as low as possible

⚠️ Still, there is the drawback mentioned before, taht the original file might be deleted if stored in /tmp.
* ✔️ the risk is low
* ✔️ I (personally) could accept that risk
* ✔️ I understand that you do neither want to add any pre/suffix, nor introduce a new 'tmp_copy_file_needs_to_be_deleted" flag.
So from my point of view, there is nothing more to do.

Many thanks for accepting and fixing my issue so quick!
🥇 

BR,
Timo
